pkg/models/models.go

```diff
@@ -0,0 +1,74 @@
+package models
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+type User struct {
+	UserID    uuid.UUID      `json:"user_id,omitempty" gorm:"type:uuid;primary_key"`
+	Email     string         `json:"email,omitempty" validate:"required,email" gorm:"unique;not null;"`
+	Username  string         `json:"username,omitempty" gorm:"unique;not null;"`
+	Password  string         `json:"password,omitempty" gorm:"not null;"`
+	Role      string         `json:"role,omitempty" gorm:"check:role='admin' or role='user'"`
+	Comments  []Comment      `json:"comments,omitempty" gorm:"foreignKey:UserID;constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
+	Posts     []Post         `json:"posts,omitempty" gorm:"foreignKey:UserID;constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
+	CreatedAt time.Time      `json:"created_at,omitempty" gorm:"autoCreateTime;"`
+	UpdatedAt time.Time      `json:"updated_at,omitempty" gorm:"autoUpdateTime;"`
+	DeletedAt gorm.DeletedAt `json:"-"`
+}
+
+type Category struct {
+	CategoryID    uuid.UUID      `json:"category_id,omitempty" gorm:"type:uuid;primary_key"`
+	Category_name string         `json:"category_name,omitempty" gorm:"unique;not null;"`
```

Comment:
* Use Pascal or Camel case for variables
```suggestion
	Category_name string         `json:"category_name,omitempty" gorm:"unique;not null;"`
	CategoryName string         `json:"category_name,omitempty" gorm:"unique;not null;"`


```